### PR TITLE
Updates to use a pipeline mapper and custom domain event transformer

### LINKED ISSUE #7: Parse timestamp/valid_at as TimeWithZone in Mapper
## Is your feature request related to a problem? Please describe.

Yes. I would like to be able to use the RES Event Browser but it fails as it expects `timestamp` to be a `Time` (or `TimeWithZone`) object as it calls `#iso8601` on it. I would also just prefer in general to interact with the event timestamp as a object vs. a string.

## Describe the solution you'd like

I think it would make sense to parse the JSON values for both `timestamp` and `valid_at` with `Time.zone.parse` in the `Mapper` class. I'm not sure if that would cause backward compatible issues though. Seems like anyone would prefer a Ruby `Time/TimeWithZone` object to an ISO date string.

This is already the case in the default YAML serializer.

## Additional context

I'm happy to create a PR if you think that would be a good step forward.

### PR DESCRIPTION
**Pull request by Samsinite**
## What changes did you make? (overview)
Updates active event store to use a pipeline mapper that uses a custom domain event transformer for dumping/loading the Active Event Store event. Fixes #7

## Is there anything you'd like reviewers to focus on?
A sanity check to validate that there is no breaking public API/event format changes.

## Checklist
- [ ] I've updates tests for this change
- [ ] I've added a Changelog entry

### CODE REVIEW

**Comment by Samsinite** [comment]
Also, I noticed that Rails Event Store also has some rspec matchers. Would you be interested in a separate PR to update the specs to use their `have_published` matcher and remove this libraries `have_published_event` matcher?

File: .gitignore [hunk 1]
```diff
@@ -29,7 +29,7 @@ log/*.log
 pkg/
 spec/internal/db/migrate
 spec/internal/db/*.sqlite3
-spec/internal/db/*.sqlite3-journal
+spec/internal/db/*.sqlite3-*
```
Comment (Samsinite) [inline .gitignore:32]: Updated to ignore the WAL/SHM sqlite DB file.

**Comment by palkan** [comment]
Thanks! Perfect work 👍 


> Would you be interested in a separate PR to update the specs to use their [have_published](https://railseventstore.org/docs/v2/rspec/#have_published) matcher and remove this library's have_published_event matcher?

I don't think `have_published` can replace `have_published_event`; the biggest different is that our matcher doesn't require explicitly specifying the store; actually, a close version would be the `publish` matcher, since our `have_published_event` only supports block targets (intentionally).

**Comment by palkan** [comment]
Released in 1.2.0

**Pull request merged**